### PR TITLE
Replace DAPS with SSI in Helm charts

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## WHAT

Changes all Helm charts to use SSI/MiW instead of DAPS.
Adds a `-legacy` helm chart that still uses DAPS, intended as a fallback.

## WHY

Replace DAPS with SSI

## FURTHER NOTES

- One chart `tractusx-connector-legacy` will be left in place that still contains DAPS for fallback purposes. However, this is _discontinued_ and will not be supported anymore. There is no deployment test for it, and it's scheduled for removal in upcoming releases.
- updated `docs/samples/example-dataspace` 
- fixed a small bug in our e2e tests, where the `MockWebServer` didn't get re-instantiated on every test

Closes # <-- _insert Issue number if one exists_

### CODE REVIEW

**Comment by paullatzelsperger** [comment]
\cc @jimmarino

File: edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+| web.http.data.path                               | X        | /data                                                                        |                            |
+| web.http.validation.port                         | X        | 8182                                                                         |                            |
+| web.http.validation.path                         | X        | /validation                                                                  |                            |
+| web.http.control.port                            | X        | 9999                                                                         |                            |
+| web.http.control.path                            | X        | /api/controlplane/control                                                    |                            |
+| web.http.ids.port                                | X        | 8282                                                                         |                            |
```
Comment (jimmarino) [inline edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md:25]: Are these `ids.` properties used anymore?

**Review by jimmarino — APPROVED** [review]

File: edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md (hunk 1 above)
Comment (paullatzelsperger) [reply edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md:25]: i don't think they are, but the README's are regenerated when we publish the helm chart. it's possible that the regenerated version never found it's way back into `main`

**Review by tuncaytunc-zf — APPROVED** [review]

File: edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md (hunk 1 above)
Comment (paullatzelsperger) [reply edc-controlplane/edc-controlplane-postgresql-hashicorp-vault-legacy/README.md:25]: also, this is only in the `legacy` chart, which will go away eventually

File: docs/samples/example-dataspace/README.md [hunk 2]
```diff
@@ -58,17 +65,19 @@ a `postStart` element to the chart's configuration file:
 ```shell
 # for sokrates
 CONFIG_FILE=sokrates-values.yaml
+CLIENT_SECRET=<sokrates-oauth-client-secret>
```
Comment (wolf4ood) [inline docs/samples/example-dataspace/README.md:69]:
```suggestion
VALUES_FILE=sokrates-values.yaml
CLIENT_SECRET=<sokrates-oauth-client-secret>
AES_KEY=$(cat sokrates.aes.key)
```

File: docs/samples/example-dataspace/README.md [hunk 3]
```diff
@@ -58,17 +65,19 @@ a `postStart` element to the chart's configuration file:
...
-yq -i ".vault.server.postStart |= [\"sh\",\"-c\",\"{\nsleep 5\n\ncat << EOF | /bin/vault kv put secret/daps-crt content=-\n$(cat sokrates.cert)\nEOF\n\n
-cat << EOF | /bin/vault kv put secret/daps-key content=-\n$(cat sokrates.key)\nEOF\n\n
-/bin/vault kv put secret/aes-keys content=$(cat sokrates.aes.key)\n\n}\"]" "$CONFIG_FILE"
+yq -i ".vault.server.postStart |= [\"sh\",\"-c\",\"{\nsleep 5\n
+/bin/vault kv put secret/client-secret content=$CLIENT_SECRET\n
+/bin/vault kv put secret/aes-keys content=$AES_KEY\n}\"]" "$VALUES_FILE"
 
 # for plato
 CONFIG_FILE=plato-values.yaml
+CLIENT_SECRET=<plato-oauth-client-secret>
```
Comment (wolf4ood) [inline docs/samples/example-dataspace/README.md:77]:
```suggestion
VALUES_FILE=plato-values.yaml
CLIENT_SECRET=<plato-oauth-client-secret>
AES_KEY=$(cat plato.aes.key)
```

**Review by wolf4ood — APPROVED** [review]
LGTM, just a minor nit on example dataspace `README`.

I've tested the chart on a local machine with a real MIW and Keycloak running.

Tested with MIW revision https://github.com/catenax-ng/tx-managed-identity-wallets/commit/bed8bd6fe25258dcaed731aeccf780611f754c14

**Comment by paullatzelsperger** [comment]
> LGTM, just a minor nit on example dataspace `README`.
> 
> I've tested the chart on a local machine with a real MIW and Keycloak running.
> 
> Tested with MIW revision [catenax-ng/tx-managed-identity-wallets@bed8bd6](https://github.com/catenax-ng/tx-managed-identity-wallets/commit/bed8bd6fe25258dcaed731aeccf780611f754c14)

Thanks @wolf4ood for doing this 🤟 I think we're on a good track then!

**Pull request merged**